### PR TITLE
Additions to the library to support the ARCore plugin

### PR DESCRIPTION
**Pull request by m4gr3d**
The ARCore plugin development is ongoing in https://github.com/GodotVR/godot_arcore

**Note:** This PR also includes the changes from #433 as the library generation code is currently broken without it.

### CODE REVIEW

File: include/core/CameraMatrix.hpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#ifndef CAMERA_MATRIX_H
```
Comment (BastiaanOlij) [inline include/core/CameraMatrix.hpp:1]: @m4gr3d we need to check if this is indeed part of the ARCore work, I don't think it is, I think this was moved or renamed at some point so we may be reintroducing something here we shouldn't

Comment (BastiaanOlij) [reply include/core/CameraMatrix.hpp:1]: Never mind, looking at the wrong project :)

**Comment by m4gr3d** [comment]
@BastiaanOlij  Is the PR good to merge?

**Comment by m4gr3d** [comment]
> @BastiaanOlij Is the PR good to merge?

@BastiaanOlij Actually nevermind, we should test the plugin and make sure it's working before starting to merge these changes.

**Comment by BastiaanOlij** [comment]
> @BastiaanOlij Actually nevermind, we should test the plugin and make sure it's working before starting to merge these changes.

Seems sensible to me, I think I've got it all compiled on my machine, just need to know how to piece it together. Mentioned it on the ARCore repo, we really need to add a small demo to the project. I'm a bit overloaded this week but if you're available in the weekend we should find an hour we can work together.

**Comment by m4gr3d** [comment]
@BastiaanOlij I've updated the README on the repo with additional instructions. They're not fully complete yet so let me know if there are any areas that need clarifications.
You had a demo project you were using for testing the ARCore integration. Can we reuse that one?

File: include/core/RID.hpp [hunk 2]
```diff
@@ -8,9 +8,10 @@ namespace godot {
 class Object;
 
 class RID {
-	godot_rid _godot_rid;
 
 public:
+	godot_rid _godot_rid;
```
[the remark this answers is not in the document]
Comment (m4gr3d) [reply include/core/RID.hpp:13]: @Zylann In the (in-development) ARCore plugin, I use it to interface with Godot ARVR APIs (e.g: [usage 1](https://github.com/GodotVR/godot_arcore/blob/405191eb2f5a16fb2a91ce86d268e4bb62341e24/plugin/src/main/cpp/arcore_interface.cpp#L385), [usage 2](https://github.com/GodotVR/godot_arcore/blob/405191eb2f5a16fb2a91ce86d268e4bb62341e24/plugin/src/main/cpp/arcore_interface.cpp#L317)).

I'm open to alternatives if any is available.

[the remark this answers is not in the document]
Comment (Zylann) [reply include/core/RID.hpp:13]: If you use C++ bindings, it's not expected to use the C functions or C structs. The ARVR API seems to have none of that wrapping, it's still C so it forces you to use the underlying structures, which is not great...

A getter seems to be enough then, but I'm confused by the functions you are using. They take `godot_rid*`, not `const godot_rid*`. Are they modifying it? Or is it an oversight?

[the remark this answers is not in the document]
Comment (m4gr3d) [reply include/core/RID.hpp]: Thanks for the tip, I've updated the implementation to use a getter instead.

**Comment by Zylann** [comment]
Thanks!

**Pull request merged**